scripts/submodule_update_with_gitee.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+#
+# Redirects git submodules to gitee mirrors and updates these recursively.
+#
+# To revert the changed URLs use 'git submodule deinit .'
+#
+
+# -----------------------------------------------------------------------------
+# Common bash
+
+if [[ ! -z ${DEBUG_SHELL} ]]
+then
+  set -x # Activate the expand mode if DEBUG is anything but empty.
+fi
+
+set -o errexit # Exit if command failed.
+set -o pipefail # Exit if pipe failed.
+set -o nounset # Exit if variable not set.
+
+die() {
+    echo "${1:-"Unknown Error"}" 1>&2
+    exit 1
+}
+
+check_git_version() {
+    currentver="$(echo $(git --version 2>&1 |awk 'NR==1{gsub(/"/,"");print $3}'))"
+    requiredver="2.11.0"
+    if [ "$(printf '%s\n' "$requiredver" "$currentver" | sort -V | head -n1)" = "$requiredver" ]; then
+        # Greater than or equal to 2.11.0
+        true
+    else
+        # Less than 2.11.0
+        false
+    fi
+}
+
+# -----------------------------------------------------------------------------
+
+ERR_CANNOT_UPDATE=13
+GITEE_GROUP_NAME="openvinotoolkit-prc"
+
+REPO_DIR=${1:-"${PWD}"}
+REPO_DIR=$(cd ${REPO_DIR} && pwd -P)
+
+SCRIPT_SH=$(cd "$(dirname "${0}")" && pwd -P)/$(basename "${0}")
+
+[ -d "${REPO_DIR}" ] || die "${REPO_DIR} is not directory!"
+[ -f "${SCRIPT_SH}" ] || die "${SCRIPT_SH} does not exist!"
+
+pushd ${REPO_DIR} >/dev/null
+
+# Step 0: Check if .gitmodules file exsit
+[ -f ".gitmodules" ] || exit 0
```

Comment:
n.b. `exit 1` since script has not performed required actions?

Reply:
`submodule_update_with_gitee.sh` will be called recursively to update submodules. If one submodule does not have `.gitmodules` file,  which means there is no sub-submodules, submodule update procedure for this repo will be skipped without error. It is desired behavior, hence `exit 0` is used. `exit 1` will stop bash script immediately if any submodules does not have sub-submodules.